dashboard/src/routes/dashboard/index.svelte

```diff
@@ -0,0 +1,152 @@
+<script lang="ts">
+	import { createDevice, getDevices } from '$lib';
+	import type { Device } from '$lib/types';
+	import {
+		authenticationData,
+		Icon,
+		ListManager,
+		NotificationManager,
+		type ActionButton,
+		type TableData
+	} from 'boxfish-studio--is-ui-components';
+	import { onMount } from 'svelte';
+	import { Container, Row } from 'sveltestrap';
+	import { DeviceDetails, ProgressBar } from '../../components';
+	import { progress } from '$lib/store';
+
+	enum State {
+		ListDevices = 'listDevices',
+		DeviceDetails = 'deviceDetails'
+	}
+
+	let devices = [];
+	let loading: boolean = false;
+	let state: State = State.ListDevices;
+	let message: string;
+	let selectedDevice: Device;
+	let query: string;
+	let searchResults: Device[] = [];
+	let CREATE_DEVICE_BUTTON: ActionButton = {
+		label: 'Create device',
+		onClick: handleCreateDevice,
+		icon: 'plus',
+		color: 'dark'
+	};
+
+	onMount(async () => {
+		await loadDevices();
+	});
+
+	async function loadDevices() {
+		devices = await getDevices($authenticationData?.did);
+		searchResults = devices;
+	}
+
+	function updateLoading(): void {
+		CREATE_DEVICE_BUTTON = {
+			icon: loading ? undefined : 'plus',
+			onClick: handleCreateDevice,
+			label: loading ? 'Creating device...' : 'Create device',
+			loading,
+			disabled: loading,
+			color: 'dark'
+		};
+	}
+
+	async function updateState(): Promise<void> {
+		if (selectedDevice) {
+			state = State.DeviceDetails;
+		} else {
+			state = State.ListDevices;
+		}
+	}
+
+	function handleSelectDevice(device: Device): void {
+		selectedDevice = device;
+	}
+
+	async function handleCreateDevice(): Promise<void> {
+		loading = true;
+		await createDevice();
+		loadDevices();
+		loading = false;
+	}
+
+	function handleBackClick(): void {
+		selectedDevice = null;
+	}
+
+	function onSearch() {
+		searchResults = devices.filter((d) => d.id?.includes(query));
+	}
+
+	$: selectedDevice, updateState();
```

Comment:
Its easier to read if all the reactive statements are on on top of the functions, imo

- imports
- props (export let)
- variables
- reactive functions
- functions